packages/matter.js/src/behavior/definitions/on-off/OnOffServer.ts

```diff
@@ -42,7 +45,7 @@ export class OnOffServer extends Base {
         await super[Symbol.asyncDispose]?.();
     }
 
-    override on() {
+    override async on() {
```

Comment:
What's the benefit of making these async?

Reply:
Ohh yes wanted to discuss that with you. As soon as someone implements these commands with real hardware behind it Theresia a high chance that he needs it Async. But then the inter cluster logic in default implementations would break. Also overriding the method in a "class extends DefaultServer" gives issues. But bigger issue is with inter cluster usage. Because lint tells me that await is not needed and such. 

So i started to implement all "interacts with a real device" command handlers as async... in factorization brings question if they should not all be async for above reasons .... ?! Wdyt?

Reply:
Hmm taking a runtime performance hit to make tooling happy is not ideal though perhaps not significant enough to worry about.  Could we manually specify MaybePromise as the return type?  Then the fact these aren't async won't affect the types.

Reply:
you mean that way?

Reply:
Adjusted to Maxbepromise ..